kiwi/system/profile.py

```diff
@@ -385,10 +393,14 @@ def _text(self, section_content):
 
     def _format(self, value):
         """
-        Helper method to format bool profile values in the way
-        the boot code expects them
+        Helper method to format None and Bool profile values in
+        the way the boot code expects them
         """
         if value is True:
             return 'true'
+        elif value is False:
+            return 'false'
+        elif not value:
+            return ''
```

Comment:
It is better, but still a bug (even unlikely will be ever triggered). :wink:  If you get non-string value, e.g. `0`, it will return "". This is not what we want (hence why I am insisting on MyPy and anno's). So as long as we want it safe enough, I would do this way (could not find a better error exception type):

```python
from typing import Optional, Any
from collections.abc import Iterable

def _format(self, value: Optional[Any]) -> str:
    """
    Helper method to format None and Bool profile values in
    the way the boot code expects them.

    Returns: String value representation
    Raises: KiwiError if value is iterable (list, dict, tuple etc).
    """
    if isinstance(value, Iterable):
        raise KiwiError("Value cannot be %s", type(value))
    elif isinstance(value, bool):
        return str(value).tolower()
    return str(value or "")
```

Also I would move this function to `utils/misc.py`.

Reply:
> if you get non-string value, e.g. 0, it will return "".

Uhh, yes you are right

Reply:
```<class 'str'>``` is also an Iterable

Reply:
so the above will raise on strings

Reply:
ok moved _format into the Shell class format_to_variable_value()